### PR TITLE
AT: adds a util function to see if at is enabled for the current site and current user

### PR DESCRIPTION
**Pull request by gwwar**
This PR adds a new util that lets us know if we should let a user start AT for the current site and current user. I'll follow up with more PRs for swapping out the current plain config check usage.

### Testing Instructions
- Navigate to http://calypso.localhost:3000/design/:siteID: (where siteID is a site that is AT ready)
- Set `localStorage.debug = 'calypso:abtests'`
- Using the bug badge abtest helper set `automatedTransfer1` to disabled
- No Upload Themes button should be shown
- Using the bug badge abtest helper set `automatedTransfer1` to enabled
- Upload Themes button should be shown

### CODE REVIEW

File: client/state/selectors/is-automated-transfer-enabled-for-site.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+	// Current User can manage site
+	const canManageSite = canCurrentUser( state, siteId, 'manage_options' );
+	if ( ! canManageSite ) {
+		return false;
+	}
+
+	// Gate to 40% roll. If we modify this value, we need a new test name
+	// in active-tests.js
+	return abtest( 'automatedTransfer1' ) === 'enabled';
```
Comment (retrofox) [inline client/state/selectors/is-automated-transfer-enabled-for-site.js]: pretty sure I'm missing something, but why are we running the test into the selector?

Comment (rralian) [inline client/state/selectors/is-automated-transfer-enabled-for-site.js]: We're only enable this for a percentage of users, so after checking all of the other checks, we also look to see whether the user is in the pool of people who are getting this variant.

**Comment by gwwar** [comment]
Wiring this in looks like a decent number of lines, going to split those out into their own PRs

**Comment by jhnstn** [comment]
non blocking question: Is there any convection on how we name selector files?
`is-automated-transfer-enabled-for-site.js` seems a bit specific to me.

**Comment by gwwar** [comment]
@jhnstn we're doing weird stuff, like an abtest roll in there, so I wanted to be more specific. Otherwise I'd wonder why it wasn't a plain flag check if it was named `is-automated-transfer-enabled`

**Comment by gwwar** [comment]
We'll need to move this to a lib and use SitesList. The tests aren't cooperating and I don't have time to track it down

**Pull request merged**